### DISCUSSION TITLE
Netflix Crashes if running MQTT

### DISCUSSION THREAD

**Discussion by Snoopy1368** [arendst/Tasmota#12958] [Q&A] [answered]
I have two of a particular brand of TV which I don't want to mention just yet but if either of the  TVs are running Netflix or a couple of other streaming services after a period of time the screen initially starts dropping frames and then will crash out with either a (-100) or a (-118) error and the following TVQ-PB-101 (5.2.5).  It is consistent and can be duplicated i.e. take the Home Assistant server off line and all is OK put it back on line and after a period of time sometimes as long as an hour, but consistently it will initially drop frames then crash.  The problem does not occur on two other well known brands of TV or if Netflix is running on a laptop or tablet.    I am new at this home automation and this is my first post so if anybody can give me some guidance with both the problem or my post I would appreciate it.  It has been suggested by some friends who use MQTT that setting up a sub net for the TV's might help, I would appreciate comments on this solution.  My significant other is a streaming addict and my life is miserable whilst her TV is crashing.

**Comment by barbudor** [comment, marked as answer]
Hi 
There could be a dozen of reasons why your setup has problems but without knowing the details of this setup it would be difficult to guide you in details.
1) Your title pointing MQTT but you stated that you shutdown the whole Home Assistant. H.A. is not only MQTT so you should shut down H.A. without shutting down MQTT to check if the problem is related to H.A. or MQTT
2) You should analyze your network traffic to understand if something is not overloading your network or creating ports conflicts. It could be H.A. itself or a specific device that goes wild when it get connected to H.A. Tools like Wireshark or tcpdump will help.
3) Segregation of networks is good indeed. 
- 1st if you have only 2.5GHz Wifi, you should consider upgrading your system to support both 2.5GHz and 5GHz and move all computer, smartphones and system that can to the 5GHz. IOT devices like ESP are using only 2.5GHz.
- Next step would be to use a separated SSID on a different Wifi channel and different network subnet to fully separate IoT from other devices.
4) if your TV is currently on Wifi but can be wired, try wire to limit Wifi congestion.

**Comment by Snoopy1368** [comment]
Firstly thanks for taking the time to give me such a detailed response obviously I should have posted more details.

 

The network whilst quite extensive does not carry much traffic I originally had an ISP provided Router / Switch and when the fault first started, I upgraded most of the hardware to Ubiquiti hoping it was a network hardware issue. The two offending TV’s are connected by Cat 5 cables not via Wi-Fi to a Ubiquiti 4 port managed switches via a Ubiquity Edge 4 Router but the managed switches have not been configured in any way.  The network is quite comprehensive consisting of a total of about 45 devices, there are two of these 4 port switches and three further Ubiquiti 8 port switches with POE.   Some of the devices are esp32’s running very simple software i.e. a ThingPulse-weather-station-color,  which polls a web site infrequently for updates. There are only two people in the dwelling although it is quite large as it was built 20 od years ago.   We are in a remote area with virtually no other houses or other Wi-Fi devices as we only have one neighbour within range.  The service is a fixed wireless and usually provide better than 50mbs.  The tablets or phones would not be used for streaming or for that matter anything at the time the TV’s would be in use.

 

The hardware is as follows.

               6 x Amazon Alexa devices in various rooms around the house 

               2 x Ubiquiti WAP’s one upstairs and one down stairs.

(the upper level is suspended on some pretty massive steel beams as the lower level is a large 4 car garage / Workshop.

               2 x Ubiquiti 4 Port Mini managed switches

               3 x 8 port Ubiquiti POE Switches.

               12 x Sonoff devices flashed with Tasmota

               6 x other ESP 32 devices not yet flashed but not Sonoff of various brands mainly light control with dimmers

               5 x ZigBee devices mainly Phillips Hue bulbs or lights.

               4 x Broadlink devices used for IR TV and Air Conditioner control and Garage Door control vie 433mhz radio.

2 x Mobile Phones Android  

               2 Tablets Android 

               1 Logitech Harmony 

               1 Sonoff Camera connected via Cat 5 Cable not Wi-Fi.

 

I have not used WireShark, I will give it a try but because in the early fault finding we included and excluded pretty much all the devices so I don’t expect it to be something going silly.   My hunch is that something is getting through the front end of the TV’s and as it does not happen to two other TV’s we have in the house I suspect it is brand specific.

 

I will proceed with separating out the TV’s on a separate network.  It will take some time as I have not done this before and am very out of touch with the technology having retired some 22 years ago and have not used Ubiquiti equipment before.

 

Thanks again for your detailed response.

 

Brian

**Comment by barbudor** [comment]
You may look into the The Hook UP videos as I think he covered segregation of networks using Ubiquiti
I would guess you are most probably facing some port conflicts than a network congestion 
On problem is that with Wireshark you will only see packets that are addressed to (or originating from) the PC running Wireshark and broadcast packets. Any other unicast packets will not been seen.
So you may have to run Wireshark/tcpdump on different devices like the server running H.A./MQTT
May be your Ubiquity router can be configured to have a monitor port on which all traffic can be copied.
Or you will need to invest into a Ethernet TAP device which allows to spy on Ethernet connections
Good luck

**Comment by Snoopy1368** [comment]
Thanks again  can you recommend an “internet TAP” device this is a new term to me, is it something similar to the Ubiquiti Cloud Key.  I will have a look at he Hook Up, I have watched a couple of his videos and found them to be helpful.

 

Brian

**Reply by barbudor** [reply]
> Thanks again can you recommend an “internet TAP” device this is a new term to me, is it something similar to the Ubiquiti Cloud Key.

Not Internet TAP, but Ethernet TAP  : https://en.wikipedia.org/wiki/Network_tap
A professional device is quite costly such as : https://www.amazon.fr/Dualcomm-Dcgs-2005L-Commutateur-Gigabit-Aliment%C3%A9/dp/B004EWVFAY
I just found that Great Scott has apparently made one that is passive : https://www.amazon.fr/Great-Scott-Gadgets-Throwing-Star/dp/B07GYVBMHZ  https://greatscottgadgets.com/throwingstar/
But this one is only for 10/100Mbps network, not Gbps

A TAP has 3 or 4 ports (device docs may use slightly different naming conventions):
- A and B for the link to spy : you have to insert the TAP across a link between 2 network element like your TV and the switch for example. The TAP will be fully transparent for all transmissions A=>B and B=>A
- X (and optionally Y) spy port : those will receive a copy of everything that goes through the A/B link. Basic ones including passive ones have X ad Y where X allows to spy A=>B and Y spies on B=>A. If you want to spy the full exchange between devices, you will need a PC with 2 ethernet ports. However in your case that might be enough to look only at 1 side at a time. More advancced (and expensives ones) will bridge X and Y spy-ports onto a single port

I've never used any of those listed above. Only used professional ones a few time at work. May be you have some friends in network that could borrow one from work for the week-end ? (and who can hep you using it)

**Comment by Snoopy1368** [comment]
Thanks for that I think I will press on at the moment with a few other options,  The manufacturer has finally taken an interest and I am happy to work with them.   We live in a geographically remote area and there is very little technology down here.  If I don’t have any luck with that I will purchase the Great Scott one, that is reasonably priced… I can easily get a second port into one of my desktops or plug two ports into my laptop but I have never tried that before.  Does not help that our dollar is very weak at the moment, we would be lucky to get $0.60 to the Euro.

 

 

 

Brian
